public/app/features/datasources/settings/CustomHeadersSettings.tsx

```diff
@@ -0,0 +1,174 @@
+import React, { PureComponent } from 'react';
+import { Button, FormField, SecretFormField } from '@grafana/ui/src';
+
+enum HeaderChangeEvent {
+  add,
+  update,
+  reset,
+  remove,
+}
+
+export interface CustomHeader {
+  name: string;
+  value: string;
+  configured: boolean;
+}
+
+export type CustomHeaders = CustomHeader[];
+
+export interface Props {
+  datasource: {
+    jsonData: {
```

Comment:
If you merge latest master you can use `KeyValue<T>` from @grafana/data package to type those props.

import { KeyValue } from '@grafana/data'